### ISSUE TITLE
bug(linter): writes to members triggering false positive for `no-unused-vars`

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#5246]
```tsx
const TestComponent = ({ testVar }) => {
  const handleClick = () => {
    testVar.index++
  }

  return <div onClick={() => handleClick()}></div>
}
```

We should try to handle this case while still reporting
```js
const foo = { a: 1 }
foo.a++
```

**Comment by DonIsaac** [comment]
uhm... why?
